registry/storage/driver/filesystem/driver.go

```diff
@@ -224,6 +224,41 @@ func (d *driver) Move(ctx context.Context, sourcePath string, destPath string) e
 	}
 
 	err := os.Rename(source, dest)
+	// os.Rename will not work across devices, in that case, try to copy then delete instead
+	if err != nil {
+		err = d.copyFile(source, dest)
```

Comment:
You can detect if the error is going across a filesystem. Read up on [`os.Rename`](https://golang.org/pkg/os/#Rename) for more details. I believe the errno is `EXDEV` but this may be different across operating systems.